login/service.go

```diff
@@ -152,6 +132,62 @@ func (gh *gitHubOAuth) Perform(ctx *app.AuthorizeLoginContext) error {
 	return ctx.TemporaryRedirect()
 }
 
+type identityError struct {
+	message interface{}
+	code    int
+}
+
+const (
+	noPrimaryEmail            int = 1
+	associatedUserNotFound    int = 2
+	unauthorizedGitHubRequest int = 3
+)
+
+func (gh *gitHubOAuth) getUserIdentity(ctx context.Context, ghtoken *oauth2.Token) (*account.Identity, *identityError) {
+	emails, err := gh.getUserEmails(ctx, ghtoken)
+	fmt.Println(emails)
+
+	primaryEmail := filterPrimaryEmail(emails)
+	if primaryEmail == "" {
+		fmt.Println("No primary email found?! ", emails)
+		return nil, &identityError{code: noPrimaryEmail, message: PrimaryEmailNotFoundError}
+	}
+	users, err := gh.users.Query(account.UserByEmails([]string{primaryEmail}), account.UserWithIdentity())
+	if err != nil {
+		return nil, &identityError{code: associatedUserNotFound, message: AssociatedUserNotFound + " " + err.Error()}
```

Comment:
Indeed, I agree that `associatedUserNotFound` is misleading, as this would probably be a network or security error instead. Yet if something wrong happens and the query fails, we should probably handle it, shouldn't we ?